src/controllers/restatedeployment/reconcilers/knative.rs

```diff
@@ -0,0 +1,938 @@
+use std::collections::BTreeMap;
+use std::time::Duration;
+
+use kube::api::{Api, Patch, PatchParams};
+use kube::{Resource, ResourceExt};
+use serde_json::json;
+use tracing::*;
+
+use crate::controllers::restatedeployment::controller::{
+    Context, RESTATE_DEPLOYMENT_ID_ANNOTATION,
+};
+use crate::controllers::restatedeployment::reconcilers::replicaset::generate_pod_template_hash;
+use crate::resources::knative::{Configuration, Revision, Route};
+use crate::resources::restatedeployments::{KnativeDeploymentStatus, RestateDeployment};
+use crate::{Error, Result};
+
+const RESTATE_TAG_ANNOTATION: &str = "restate.dev/tag";
+const RESTATE_REGISTERED_AT_ANNOTATION: &str = "restate.dev/registered-at";
+const RESTATE_DEPLOYMENT_ANNOTATION: &str = "restate.dev/deployment";
+
+/// Main Knative reconciliation function
+/// Updates status in-place and returns the next_removal time for requeue logic
+pub async fn reconcile_knative(
+    ctx: &Context,
+    rsd: &RestateDeployment,
+    status: &mut crate::resources::restatedeployments::RestateDeploymentStatus,
+) -> Result<Option<chrono::DateTime<chrono::Utc>>> {
+    let namespace = rsd
+        .namespace()
+        .ok_or_else(|| Error::InvalidRestateDeployment("RestateDeployment must have a namespace".into()))?;
+
+    info!(
+        namespace = %namespace,
+        name = %rsd.name_any(),
+        "Reconciling Knative deployment"
+    );
+
+    // Step 1: Determine current tag
+    let current_tag = determine_tag(rsd)?;
+    info!(tag = %current_tag, "Determined deployment tag");
+
+    // Step 2: Reconcile Configuration for current tag
+    let config = reconcile_configuration(ctx, rsd, &namespace, &current_tag).await?;
+    info!(configuration = %config.name_any(), "Configuration reconciled");
+
+    // Step 3: Reconcile Route for current tag
+    let route = reconcile_route(ctx, rsd, &namespace, &current_tag, &config).await?;
+    info!(route = %route.name_any(), "Route reconciled");
+
+    // Step 4: Get the latest created revision (observe rollout eagerly)
+    let latest_revision = config
+        .status
+        .as_ref()
+        .and_then(|s| s.latest_created_revision_name.as_ref())
+        .ok_or_else(|| Error::ConfigurationNotReady {
+            message: format!(
+                "Configuration {} does not have latestCreatedRevisionName in status",
+                config.name_any()
+            ),
+            reason: "RevisionNotCreated".into(),
+            requeue_after: Some(Duration::from_secs(5)),
+        })?
+        .clone();
+    info!(revision = %latest_revision, "Latest revision created");
+
+    // Fetch the full Revision object for replica counts
+    let revision_api: Api<Revision> = Api::namespaced(ctx.client.clone(), &namespace);
+    let revision = revision_api.get(&latest_revision).await?;
```

Comment:
elsewhere in this operator, when we do an explicit get, we use a reflector, can we do the same here? so youd have to pass down the Store managed by the reflector like we do for replicaset